gemini/sample-apps/customer-search/functions/create_fd/main.py

```diff
@@ -0,0 +1,230 @@
+      number_of_days: The number of days in the fixed deposit tenure.
+
+    Returns:
+      The interest rate for the fixed deposit as a float.
+    """
+
+    client = bigquery.Client()
+
+    roi = 3.0
+
+    person = "rate_of_interest"
+    if is_sr_citizen:
+        person = "rate_of_interest_sr_citizen"
+
+    query_interest_rate = f"""
+SELECT {person} as rate_of_interest FROM `{project_id}.DummyBankDataset.FdInterestRates` \
+where bucket_start_days < {number_of_days} order by bucket_start_days desc limit 1
+"""
+
+    result_query_interest_rate = client.query(query_interest_rate)
+
+    for row in result_query_interest_rate:
+        if row["rate_of_interest"] is not None:
+            roi = row["rate_of_interest"]
+
+    return roi
+
+
+@functions_framework.http
+def create_fixed_deposit(request):
+    """
+    Creates a fixed deposit for a customer.
+
+    Args:
+        request (flask.Request): The request object.
+            <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
+
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+    """
+
+    request_json = request.get_json(silent=True)
+
+    customer_id = request_json["sessionInfo"]["parameters"]["cust_id"]
+    fd_amount = request_json["sessionInfo"]["parameters"]["fd_amount"]
+    fd_tenure = request_json["sessionInfo"]["parameters"]["fd_tenure"]
+    user_name = request_json["sessionInfo"]["parameters"]["name"]
+
+    query_handler = BigQueryHandler(customer_id=customer_id)
+
+    # verifying that the customer is valid and exists in our database or not
+    cust_id_exists, res = query_handler.validate_customer_id()
+    if not cust_id_exists:
+        return res
+
+    # get the date of birth of the user
+    result_dob = query_handler.query("query_dob")
+
+    is_sr_citizen = False
+
+    for row in result_dob:
+        if row["dob"] is not None:
+            dob = row["dob"]
+            is_sr_citizen = check_senior_citizen(dob)
+
+    number_of_days = get_number_of_days(fd_tenure)
+    present_date = date.today()
+    present_date = present_date.isoformat()
+    interest_rate = get_interest_rate(is_sr_citizen, number_of_days)
+
+    model = Gemini()
+
+    response = model.generate_response(
+        f"""Generate a confirmation message on creating a fd.
+    The user name is {user_name}
+    The amount is {fd_amount}
+    The tenure is {fd_tenure}
+    The interest rate is {interest_rate} .
+    The order should be amount then tenure then interest rate and these things
+    should be in bullet points.
+    The bank name is Cymbal Bank.
+    The amount should be in the format ₹1,20,312.15 instead of 120312.15
+    Do not show maturity date and maturity amount.
+    The output should be within 40 words. Use emoji.
+    """
+    )
```

Comment:
I'm not sure this is the best way to write this. You could just use a template and fill in the values.